### ISSUE TITLE
Add compound input list for retention time calibration 

### ISSUE THREAD

**Issue by ansgarkorf** [mzmine/mzmine#1557]
Current implementation of the retention time calibration module searches for a common feature in multiple feature lists. After successful calibration, no info on the utilized feature can be found. Therefore, allow manual definition of compounds used for retention time calibration.  

- [ ] Add RT corrected data type to features and rows (bind)
- [ ] Advanced parameter in alignment - use corrected RT (default)

**Comment by tomas-pluskal** [comment]
I think the current module should be discarded and we should add RT calibration at the raw data level. This is what other platforms do (e.g. XCMS, Progenesis) and it has the advantage that the user can immediately see the aligned peaks in the TIC plot.

**Comment by tomas-pluskal** [comment]
We could consider this approach: https://www.nature.com/articles/s41467-023-43909-5

**Comment by tomas-pluskal** [comment]
Yes, Progenesis definitely modifies the retention times of the MS1 scans before performing feature detection. I think XCMS does that too, although not 100% sure. This approach makes gap-filling much easier later, because the RT range of each feature is constant across samples. It also makes alignment easier.

**Comment by tomas-pluskal** [comment]
This is from XMCS docs:
```
The adjustRtime method(s) perform retention time correction (alignment) between chromatograms of different samples. Alignment is performed by default on MS level 1 data. Retention times of spec- tra from other MS levels, if present, are subsequently adjusted based on the adjusted retention times of the MS1 spectra. Note that calling adjustRtime on a xcms result object will remove any even- tually present previous alignment results as well as any correspondence analysis results. To run a second round of alignment, raw retention times need to be replaced with adjusted ones using the applyAdjustedRtime() function.
```

**Comment by robinschmid** [comment]
So it sounds like they overwrite the retention time in the scan headers.

**Comment by SteffenHeu** [comment]
Approaches:

**Correct RT on the feature level:**
- [ ] just replace the RTType with the corrected retention time 
  - Shapes and RT will mismatch, change of IonTimeSeries will cause FeatureDataUtils to replace the corrected RT
  - No changes needed in any of the module codes, as the corrected RT will be used by default
  - RT and representative spectrum will have different retention times

- [ ] create a second type CorrectedRTType 
  - Shapes and RT will match, shapes and CorrectedRT will mismatch
  - Adjustments needed to downstream tools such as Speclib search, DB search, Alignment, gap filling to either use corrected RT or original RT.
    - Corrected: Alignment, DB + Speclib searches, filtering
    - Original: Gap filling
- Comparability with other software that shows raw data is still there (same RTs)

**Correct RT on the raw data level**
- No issues with two RT types or downstream processing modules
- possibility for negative retention times, which will mess up feature area calculation
- no comparability of retention times with other software
- against the philosophy of unaltered raw data

**Comment by robinschmid** [comment]
Might be good to try out the corrected RT in the scan header. This would be the easiest fix actually. 
We could just add methods to Scan:
- getRawRetentionTime
- isRetentionTimeCorrected

And the old getRetentionTime method would return the corrected RT if present or the original RT otherwise. 

As for negative RT values, this could be restricted by capping it at 0 or shifting to positive?

**Comment by SteffenHeu** [comment]
done
#2476
